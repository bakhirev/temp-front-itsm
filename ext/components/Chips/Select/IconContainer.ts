import styled, { css } from 'styled-components';

import { DEFAULT_THEME } from '../../common';

import { getIconColor } from './styleUtils';

export interface IIconContainerProps {
  disabled?: boolean;
  selected?: boolean;
}

export const IconContainer = styled.div<IIconContainerProps>`
  ${({ theme, disabled, selected }) => css`
    display: flex;
    justify-content: center;
    & svg {
      fill: ${getIconColor(theme, selected, disabled)};
    }
  `}
`;

IconContainer.defaultProps = {
  theme: DEFAULT_THEME,
};
