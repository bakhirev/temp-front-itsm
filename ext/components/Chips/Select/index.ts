export { Select, IChipsSelectProps } from './Select';
export { MultiSelect, IChipsMultiSelectProps } from './MultiSelect';
export { IItemChipSelect } from './types';
