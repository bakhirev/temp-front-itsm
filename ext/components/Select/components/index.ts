export { Menu, MenuList } from './Menu';
export { DropdownIndicator, ClearIndicator, IndicatorsContainer } from './Indicator';
export { Option, NoOptionsMessage } from './Option';
export { OptionProduct } from './OptionProduct';
export { ValueContainer, SingleValue } from './Value';
export { SelectContainerComponent, ControlComponent, Placeholder } from './Others';
