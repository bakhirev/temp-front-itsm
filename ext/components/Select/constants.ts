import { ReactNode } from 'react';

export type SelectType = 'default' | 'product' | 'card';
export type Size = 'big' | 'small' | 'micro';

export const VALUE_CONTAINER_MARGIN_RIGHT = '8px';
export const LABEL_MARGIN_BOTTOM = '8px';
export const ADDITIONAL_TEXT_MARGIN_TOP = '8px';
export const BOTTOM_LINE_HEIGHT_ON_FOCUS = '2px';
export const SELECT_DEFAULT_WIDTH = '320px';
export const CLOSE_ICON_MARGIN_RIGHT = '4px';
export const ICON_DEFAULT_SIZE = 24;
export const ICON_MICRO_SIZE = 20;

export const ICON_MARGIN_RIGHT_DEFAULT = '12px';
export const ICON_MARGIN_RIGHT_MICRO = '8px';

export const CONTROL_HEIGHT = {
  micro: '32px',
  small: '40px',
  big: '56px',
};
export const OPTION_HEIGHT = {
  micro: '32px',
  small: '40px',
  big: '48px',
};

export const OPTION_PADDING = {
  micro: '6px 12px',
  small: '8px 16px',
  big: '12px 16px',
};
export const CONTROL_PADDING = (type = 'default') => ({
  micro: '6px 11px',
  small: '8px 15px',
  big: `${type === 'default' ? '12px 15px' : '10px 15px'}`,
});

export const MAX_LINES_AMOUNT_IN_MENU = 8;
export const MENU_VERTICAL_PADDING = '8px';
export const MENU_MAX_HEIGHT = {
  micro: `${
    parseInt(OPTION_HEIGHT.micro) * MAX_LINES_AMOUNT_IN_MENU + parseInt(MENU_VERTICAL_PADDING) * 2
  }px`,
  small: `${
    parseInt(OPTION_HEIGHT.small) * MAX_LINES_AMOUNT_IN_MENU + parseInt(MENU_VERTICAL_PADDING) * 2
  }px`,
  big: `${
    parseInt(OPTION_HEIGHT.big) * MAX_LINES_AMOUNT_IN_MENU + parseInt(MENU_VERTICAL_PADDING) * 2
  }px`,
};

export type SelectCurrency = 'usd' | 'eur' | 'rur';

export const currencyToSymbol: {
  [key in SelectCurrency]: string | ReactNode;
} = {
  usd: '$',
  eur: '€',
  rur: '₽',
};

export const PRODUCT_OPTION_ICON_SPACING = 12;

export const PRODUCT_OPTION_ICON = {
  product: { width: 32, height: 32, radius: '50%' },
  card: { width: 44, height: 28, radius: '2px' },
};
