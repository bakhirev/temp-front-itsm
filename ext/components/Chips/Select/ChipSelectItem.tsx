import styled, { css } from 'styled-components';

import { DEFAULT_THEME } from '../../common';
import { HEIGHT_BIG, HEIGHT_SMALL, MAX_WIDTH, SHAPE_MICRO } from '../common/constants';
import type { Size } from '../Tags/ChipTagItem';

import {
  getChipColor,
  getChipBackgroundColor,
  getChipHoverColor,
  getChipHoverBackgroundColor,
} from './styleUtils';

export interface IChipSelectItemProps {
  size: Size;
  disabled?: boolean;
  selected?: boolean;
  focused?: boolean;
}

export const ChipSelectItem = styled.div<IChipSelectItemProps>`
  ${({ theme, disabled, size, selected }) => css`
    display: flex;
    align-items: center;
    position: relative;
    cursor: ${disabled ? 'default' : 'pointer'};
    padding-left: 8px;
    padding-right: 8px;
    max-width: ${MAX_WIDTH};
    color: ${getChipColor(theme, disabled, selected)};
    background-color: ${getChipBackgroundColor(theme, disabled, selected)};
    border-radius: ${SHAPE_MICRO};
    height: ${size === 'big' ? HEIGHT_BIG : HEIGHT_SMALL};
    ${disabled
      ? ''
      : css`
          &:hover {
            color: ${getChipHoverColor(theme, selected)};
            background-color: ${getChipHoverBackgroundColor(theme, selected)};
          }
        `}
    &:focus {
      outline: none;
      ${disabled ? '' : `border: 2px solid ${theme.color.primary[60]};`}
    }
  `}
`;

ChipSelectItem.defaultProps = {
  theme: DEFAULT_THEME,
};
