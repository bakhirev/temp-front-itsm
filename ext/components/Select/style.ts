import { OPTION_PADDING } from './constants';

export const colorStyles = ({ size }: any) => {
  return {
    dropdownIndicator: (styles: any) => {
      return {
        ...styles,
        display: 'flex',
      };
    },
    indicatorSeparator: () => ({}),
    noOptionsMessage: (base: any) => ({
      ...base,
      textAlign: 'left',
      padding: OPTION_PADDING[size],
    }),
    input: (styles: any) => ({
      ...styles,
      marginLeft: '0 !important',
    }),
  };
};
