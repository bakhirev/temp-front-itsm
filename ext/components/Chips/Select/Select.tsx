import React, { Fragment, FC, useState } from 'react';

import { useTooltip } from '../../common/hooks';
import { createTooltipHOC } from '../../TooltipHOC';
import { KEY_CODES } from '../../common';
import { ICON_SIZE_SMALL, ICON_SIZE_BIG } from '../common/constants';
import { ChipsWrapper } from '../common/ChipsWrapper';
import { Separator } from '../common/Separator';
import type { Size } from '../common/common';
import { StyledCaption11 } from '../common/Caption11';
import { Caption13 } from '../common/Caption13';

import { IItemChipSelect } from './types';
import { ChipSelectItem } from './ChipSelectItem';
import { IconContainer } from './IconContainer';
import { SelectLabelContainer } from './SelectLabelContainer';

export interface IChipsSelectProps {
  /** Массив чипсов */
  items: IItemChipSelect[];
  /** Размер чипсов */
  size?: Size;
  /** Идентификатор выбранного элемента */
  selectedId?: string;
  /** Обработчик выбора элемента */
  onSelectItem?: (id: string) => void;
  /** Обработчик фокусировки на элементе */
  onFocusItem?: (id: string) => void;
  /** В селекте или нет, влияет на размер отступа */
  inSelect?: boolean;
  /** Отключение чипсов */
  disabled?: boolean;
  /** Имя класса для переопределения стилей */
  className?: string;
  /** data-test-id атрибут для тестирования компонента */
  dataTestId?: string;
  /** Ширина Тега */
  width?: number | string;
  /** Контейнер, в котором при необходимости можно отрисовать тултип через ReactDOM.createPortal */
  tooltipContainer?: Element | null;
}

export const Select: FC<IChipsSelectProps> = ({
  items,
  inSelect,
  className,
  dataTestId,
  size = 'big',
  selectedId,
  onSelectItem,
  onFocusItem,
  width,
  tooltipContainer,
}) => {
  const [tooltipItems, itemsRef] = useTooltip();
  const [focusedItem, setFocusedItem] = useState(null);
  const [selectedIdState, setSelectedIdState] = useState(
    selectedId || items.length ? items[0].id : null
  );
  const handleFocus = (itemId) => {
    setFocusedItem(itemId);
    onFocusItem && onFocusItem(itemId);
  };
  const handleItemSelect = (itemId) => {
    setSelectedIdState(itemId);
    onSelectItem && onSelectItem(itemId);
  };

  const onKeyDownCreator = (itemId) => (event) => {
    if ([KEY_CODES.SPACEBAR, KEY_CODES.ENTER].includes(event.keyCode)) {
      event.preventDefault();
      handleItemSelect(itemId);
    }
  };

  return (
    <ChipsWrapper className={className} data-test-id={dataTestId}>
      {items.map((chipItem, index) => {
        const Chip = tooltipItems[chipItem.id] ? createTooltipHOC(ChipSelectItem) : ChipSelectItem;
        const Label = size === 'small' ? StyledCaption11 : Caption13;
        const iconSize = size === 'big' ? ICON_SIZE_BIG : ICON_SIZE_SMALL;
        const Icon = chipItem.icon;

        return (
          <Fragment key={chipItem.id}>
            <Chip
              tooltip={chipItem.label}
              container={tooltipContainer}
              size={size}
              disabled={chipItem.disabled}
              onClick={chipItem.disabled ? undefined : () => handleItemSelect(chipItem.id)}
              selected={selectedIdState === chipItem.id}
              onFocus={chipItem.disabled ? undefined : () => handleFocus(chipItem.id)}
              focused={focusedItem === chipItem.id}
              onKeyDown={chipItem.disabled ? undefined : onKeyDownCreator(chipItem.id)}
              tabIndex={chipItem.disabled ? -1 : 1}
            >
              {Icon && (
                <IconContainer
                  disabled={chipItem.disabled}
                  selected={selectedIdState === chipItem.id}
                >
                  <Icon width={iconSize} height={iconSize} />
                </IconContainer>
              )}
              <SelectLabelContainer size={size} width={width} withIcon={!!Icon}>
                <Label ref={(el) => itemsRef(el, chipItem.id)}>{chipItem.label}</Label>
              </SelectLabelContainer>
            </Chip>
            {index !== items.length - 1 && <Separator inSelect={inSelect} />}
          </Fragment>
        );
      })}
    </ChipsWrapper>
  );
};
