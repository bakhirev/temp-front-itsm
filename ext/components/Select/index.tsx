import React, { ReactNode, FC, useRef } from 'react';
import ReactSelect from 'react-select';

import { colorStyles } from './style';
import {
  ClearIndicator,
  ControlComponent,
  DropdownIndicator,
  IndicatorsContainer,
  Menu,
  MenuList,
  NoOptionsMessage,
  Option,
  OptionProduct,
  Placeholder,
  SelectContainerComponent,
  SingleValue,
  ValueContainer,
} from './components';
import * as SelectConstants from './constants';

export interface ISelectComponentItem<T = any> {
  /** Подпись к опции */
  label: string | number;
  /** Уникальное значение опции */
  value: T;
  /** Выбор опции */
  isSelected?: boolean;
  /** Отключение опции */
  isDisabled?: boolean;
  /** Фокус опции */
  isFocused?: boolean;
  /** Иконка для опции. Используется в обычном селекте */
  icon?: ReactNode;
  /** Подпись для продукта и карты */
  note?: ReactNode;
  /** Путь до изображения продукта / карты */
  image?: ReactNode;
  /** Добавить знак валюты */
  currency?: string | SelectConstants.SelectCurrency;
}

export interface ISelectComponentProps {
  /** Список опций */
  list: ISelectComponentItem[];
  /** Коллбэк на изменение значения селекта */
  onChange?: (value: any) => void;
  /** Коллбэк при потере селектом фокуса */
  onBlur?: (value: any) => void;
  /** Коллбэк при установке фокуса на селект */
  onFocus?: (e: any) => void;
  /** Уникальное имя инпута */
  name?: string;
  /** Верхняя надпись к селекту */
  label?: string;
  /** Плейсхолдер селекта */
  placeholder?: string;
  /** Дополнительный текст под селектом */
  additionalText?: string;
  /** Ширина селекта */
  width?: string | number;
  /** Размер компонента */
  size?: SelectConstants.Size;
  /** Отключение компонента */
  disabled?: boolean;
  /** Открытие меню */
  menuIsOpen?: boolean;
  /** Селект с возможностью очищения, по умолчанию false */
  clearable?: boolean;
  /** Селект в состоянии ошибки */
  error?: boolean;
  /** Дефолтное значение селекта */
  defaultValue?: ISelectComponentItem;
  /** Значение селекта */
  value?: ISelectComponentItem;
  /** Сообщение, которое будет выведено в меню при отсутствии опций */
  withoutOptionMessage?: string;
  /** Имя класса для переопределения стилей */
  className?: string;
  /** data-test-id атрибут для тестирования компонента */
  dataTestId?: string;
  /** Тип селекта */
  type?: SelectConstants.SelectType;
  /** Должен ли происходить скролл страницы, если при открытии меню выпадающему списку не хватает места */
  menuShouldScrollIntoView?: boolean;
  /** Контейнер, в котором при необходимости можно отрисовать тултип через ReactDOM.createPortal */
  tooltipContainer?: Element | null;
}

export const Select: FC<ISelectComponentProps> = (props) => {
  const {
    placeholder = '',
    label,
    disabled,
    list,
    menuIsOpen,
    onChange,
    defaultValue,
    clearable = false,
    withoutOptionMessage = 'Варианты не найдены',
    size = 'big',
    additionalText,
    error = false,
    width = SelectConstants.SELECT_DEFAULT_WIDTH,
    name,
    value,
    onFocus,
    onBlur,
    className,
    dataTestId,
    type = 'default',
    menuShouldScrollIntoView = false,
    tooltipContainer,
  } = props;

  const isProductOrCardType = ['card', 'product'].includes(type);
  const tooltipPlaceholderRef = useRef(null);

  return (
    <ReactSelect
      components={{
        SelectContainer: SelectContainerComponent,
        Menu,
        IndicatorsContainer,
        DropdownIndicator,
        ClearIndicator,
        NoOptionsMessage,
        Option: isProductOrCardType ? OptionProduct : Option,
        Control: ControlComponent,
        Placeholder,
        ValueContainer,
        MenuList,
        SingleValue,
      }}
      menuShouldScrollIntoView={menuShouldScrollIntoView}
      isSearchable={false}
      label={label}
      name={name}
      value={value}
      defaultValue={defaultValue}
      menuIsOpen={!disabled && menuIsOpen}
      options={list}
      styles={colorStyles(props)}
      isDisabled={disabled}
      placeholder={placeholder}
      onChange={(item) => onChange && onChange(item && (item as ISelectComponentItem).value)}
      onFocus={onFocus}
      onBlur={onBlur}
      isClearable={clearable}
      withoutOptionMessage={withoutOptionMessage}
      size={isProductOrCardType ? 'big' : size}
      error={error}
      additionalText={additionalText}
      width={width}
      className={className}
      dataTestId={dataTestId}
      type={type}
      tooltipPlaceholderRef={tooltipPlaceholderRef}
      tooltipContainer={tooltipContainer}
    />
  );
};

export { SelectConstants };
