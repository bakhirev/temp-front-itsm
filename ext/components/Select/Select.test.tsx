import React from 'react';
import { shallow } from 'enzyme';

import { ReactComponent as PersonOutline } from '../Icons/system/PersonOutline.svg';
import { ReactComponent as SaveOutline } from '../Icons/system/SaveOutline.svg';
import { ReactComponent as PrintOutline } from '../Icons/system/PrintOutline.svg';
import { Select, ISelectComponentProps } from '../Select';
jest.mock('../common', () => ({
  DEFAULT_THEME: {},
  Z_INDEXES: {
    TOOLTIP: 777,
  },
}));

describe('Select', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const selectRequiredProps: ISelectComponentProps = {
    list: [
      { value: 1, label: 'Москва' },
      { value: 2, label: 'Новгород' },
      { value: 3, label: 'Орел' },
    ],
  };

  const selectWithIconRequiredProps: ISelectComponentProps = {
    list: [
      { value: 1, label: 'Москва', icon: PersonOutline },
      { value: 2, label: 'Новгород', icon: SaveOutline },
      { value: 3, label: 'Орел', icon: PrintOutline },
    ],
  };

  const cardOrProductSelectProps: ISelectComponentProps = {
    list: [
      {
        value: 17812,
        label: '123000.56',
        note: 'Дебетовая карта (*7812)',
        // eslint-disable-next-line react/display-name
        image: () => <div />,
        currency: 'rur',
      },
      {
        value: 298,
        label: '905.00',
        note: 'Кредитная карта (*1234)',
        // eslint-disable-next-line react/display-name
        image: () => <div />,
        currency: 'usd',
      },
      {
        value: 334,
        label: '7775.00',
        note: 'Дебетовая карта (*9845)',
        // eslint-disable-next-line react/display-name
        image: () => <div />,
        currency: 'eur',
      },
    ],
  };

  it('should render component with required properties', () => {
    const wrapper = shallow(<Select {...selectRequiredProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with required properties and icon', () => {
    const wrapper = shallow(<Select {...selectWithIconRequiredProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with small size', () => {
    const wrapper = shallow(<Select {...selectRequiredProps} size="small" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with micro size', () => {
    const wrapper = shallow(<Select {...selectRequiredProps} size="micro" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with disabled state', () => {
    const wrapper = shallow(<Select {...selectRequiredProps} disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with error state', () => {
    const wrapper = shallow(<Select {...selectRequiredProps} error />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with label, placeholder, additionalText', () => {
    const wrapper = shallow(
      <Select
        {...selectRequiredProps}
        label="Город"
        placeholder="Выберите из списка"
        additionalText="Место фактического проживания"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with card type', () => {
    const wrapper = shallow(<Select {...cardOrProductSelectProps} type="card" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with product type', () => {
    const wrapper = shallow(<Select {...cardOrProductSelectProps} type="product" />);
    expect(wrapper).toMatchSnapshot();
  });
});
