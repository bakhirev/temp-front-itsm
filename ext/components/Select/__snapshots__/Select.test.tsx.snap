// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render component with card type 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": [Function],
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "NoOptionsMessage": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  error={false}
  isClearable={false}
  isSearchable={false}
  menuShouldScrollIntoView={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "currency": "rur",
        "image": [Function],
        "label": "123000.56",
        "note": "Дебетовая карта (*7812)",
        "value": 17812,
      },
      Object {
        "currency": "usd",
        "image": [Function],
        "label": "905.00",
        "note": "Кредитная карта (*1234)",
        "value": 298,
      },
      Object {
        "currency": "eur",
        "image": [Function],
        "label": "7775.00",
        "note": "Дебетовая карта (*9845)",
        "value": 334,
      },
    ]
  }
  placeholder=""
  size="big"
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "input": [Function],
      "noOptionsMessage": [Function],
    }
  }
  tooltipPlaceholderRef={
    Object {
      "current": null,
    }
  }
  type="card"
  width="320px"
  withoutOptionMessage="Варианты не найдены"
/>
`;

exports[`Select should render component with disabled state 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": [Function],
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "NoOptionsMessage": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  error={false}
  isClearable={false}
  isDisabled={true}
  isSearchable={false}
  menuIsOpen={false}
  menuShouldScrollIntoView={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  placeholder=""
  size="big"
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "input": [Function],
      "noOptionsMessage": [Function],
    }
  }
  tooltipPlaceholderRef={
    Object {
      "current": null,
    }
  }
  type="default"
  width="320px"
  withoutOptionMessage="Варианты не найдены"
/>
`;

exports[`Select should render component with error state 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": [Function],
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "NoOptionsMessage": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  error={true}
  isClearable={false}
  isSearchable={false}
  menuShouldScrollIntoView={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  placeholder=""
  size="big"
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "input": [Function],
      "noOptionsMessage": [Function],
    }
  }
  tooltipPlaceholderRef={
    Object {
      "current": null,
    }
  }
  type="default"
  width="320px"
  withoutOptionMessage="Варианты не найдены"
/>
`;

exports[`Select should render component with label, placeholder, additionalText 1`] = `
<StateManager
  additionalText="Место фактического проживания"
  components={
    Object {
      "ClearIndicator": [Function],
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "NoOptionsMessage": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  error={false}
  isClearable={false}
  isSearchable={false}
  label="Город"
  menuShouldScrollIntoView={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  placeholder="Выберите из списка"
  size="big"
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "input": [Function],
      "noOptionsMessage": [Function],
    }
  }
  tooltipPlaceholderRef={
    Object {
      "current": null,
    }
  }
  type="default"
  width="320px"
  withoutOptionMessage="Варианты не найдены"
/>
`;

exports[`Select should render component with micro size 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": [Function],
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "NoOptionsMessage": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  error={false}
  isClearable={false}
  isSearchable={false}
  menuShouldScrollIntoView={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  placeholder=""
  size="micro"
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "input": [Function],
      "noOptionsMessage": [Function],
    }
  }
  tooltipPlaceholderRef={
    Object {
      "current": null,
    }
  }
  type="default"
  width="320px"
  withoutOptionMessage="Варианты не найдены"
/>
`;

exports[`Select should render component with product type 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": [Function],
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "NoOptionsMessage": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  error={false}
  isClearable={false}
  isSearchable={false}
  menuShouldScrollIntoView={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "currency": "rur",
        "image": [Function],
        "label": "123000.56",
        "note": "Дебетовая карта (*7812)",
        "value": 17812,
      },
      Object {
        "currency": "usd",
        "image": [Function],
        "label": "905.00",
        "note": "Кредитная карта (*1234)",
        "value": 298,
      },
      Object {
        "currency": "eur",
        "image": [Function],
        "label": "7775.00",
        "note": "Дебетовая карта (*9845)",
        "value": 334,
      },
    ]
  }
  placeholder=""
  size="big"
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "input": [Function],
      "noOptionsMessage": [Function],
    }
  }
  tooltipPlaceholderRef={
    Object {
      "current": null,
    }
  }
  type="product"
  width="320px"
  withoutOptionMessage="Варианты не найдены"
/>
`;

exports[`Select should render component with required properties 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": [Function],
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "NoOptionsMessage": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  error={false}
  isClearable={false}
  isSearchable={false}
  menuShouldScrollIntoView={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  placeholder=""
  size="big"
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "input": [Function],
      "noOptionsMessage": [Function],
    }
  }
  tooltipPlaceholderRef={
    Object {
      "current": null,
    }
  }
  type="default"
  width="320px"
  withoutOptionMessage="Варианты не найдены"
/>
`;

exports[`Select should render component with required properties and icon 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": [Function],
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "NoOptionsMessage": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  error={false}
  isClearable={false}
  isSearchable={false}
  menuShouldScrollIntoView={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "icon": "div",
        "label": "Москва",
        "value": 1,
      },
      Object {
        "icon": "div",
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "icon": "div",
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  placeholder=""
  size="big"
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "input": [Function],
      "noOptionsMessage": [Function],
    }
  }
  tooltipPlaceholderRef={
    Object {
      "current": null,
    }
  }
  type="default"
  width="320px"
  withoutOptionMessage="Варианты не найдены"
/>
`;

exports[`Select should render component with small size 1`] = `
<StateManager
  components={
    Object {
      "ClearIndicator": [Function],
      "Control": [Function],
      "DropdownIndicator": [Function],
      "IndicatorsContainer": [Function],
      "Menu": [Function],
      "MenuList": [Function],
      "NoOptionsMessage": [Function],
      "Option": [Function],
      "Placeholder": [Function],
      "SelectContainer": [Function],
      "SingleValue": [Function],
      "ValueContainer": [Function],
    }
  }
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  error={false}
  isClearable={false}
  isSearchable={false}
  menuShouldScrollIntoView={false}
  onChange={[Function]}
  options={
    Array [
      Object {
        "label": "Москва",
        "value": 1,
      },
      Object {
        "label": "Новгород",
        "value": 2,
      },
      Object {
        "label": "Орел",
        "value": 3,
      },
    ]
  }
  placeholder=""
  size="small"
  styles={
    Object {
      "dropdownIndicator": [Function],
      "indicatorSeparator": [Function],
      "input": [Function],
      "noOptionsMessage": [Function],
    }
  }
  tooltipPlaceholderRef={
    Object {
      "current": null,
    }
  }
  type="default"
  width="320px"
  withoutOptionMessage="Варианты не найдены"
/>
`;
